dbt/adapters/databricks/python_models/python_submissions.py

```diff
@@ -138,189 +224,318 @@ def _submit_job(self, path: str, cluster_spec: dict) -> str:
-        return run_id
-
-    def _build_additional_job_settings(self) -> Dict[str, Any]:
-        additional_configs = {}
-        attrs_to_add = [
-            "email_notifications",
-            "webhook_notifications",
-            "notification_settings",
-            "timeout_seconds",
-            "health",
-            "environments",
-        ]
-        for attr in attrs_to_add:
-            if attr in self.workflow_spec:
-                additional_configs[attr] = self.workflow_spec[attr]
-
-        return additional_configs
-
-    def _submit_through_notebook(self, compiled_code: str, cluster_spec: dict) -> None:
-        workdir = self.api_client.workspace.create_python_model_dir(
-            self.database or "hive_metastore", self.schema
+        job_spec: Dict[str, Any] = {
+            "task_key": "inner_notebook",
+            "notebook_task": {
+                "notebook_path": path,
+            },
+        }
+        job_spec.update(self.cluster_spec)  # updates 'new_cluster' config
+
+        additional_job_config = self.additional_job_settings
+        access_control_list = self.permission_builder.build_job_permissions()
+        if access_control_list:
+            job_spec["access_control_list"] = access_control_list
+
+        return PythonJobDetails(self.run_name, job_spec, additional_job_config)
+
+
+class PythonNotebookSubmitter(PythonSubmitter):
+    """Submitter for Python models using the Job API."""
+
+    def __init__(
+        self,
+        api_client: DatabricksApiClient,
+        tracker: PythonRunTracker,
+        uploader: PythonNotebookUploader,
+        config_compiler: PythonJobConfigCompiler,
+    ) -> None:
+        self.api_client = api_client
+        self.tracker = tracker
+        self.uploader = uploader
+        self.config_compiler = config_compiler
+
+    @staticmethod
+    def create(
+        api_client: DatabricksApiClient,
+        tracker: PythonRunTracker,
+        parsed_model: ParsedPythonModel,
+        cluster_spec: Dict[str, Any],
+    ) -> "PythonNotebookSubmitter":
+        notebook_uploader = PythonNotebookUploader.create(api_client, parsed_model)
+        config_compiler = PythonJobConfigCompiler.create(
+            api_client,
+            parsed_model,
+            cluster_spec,
         )
-        file_path = f"{workdir}{self.identifier}"
+        return PythonNotebookSubmitter(api_client, tracker, notebook_uploader, config_compiler)
 
-        self.api_client.workspace.upload_notebook(file_path, compiled_code)
+    @override
+    def submit(self, compiled_code: str) -> None:
+        file_path = self.uploader.upload(compiled_code)
+        job_config = self.config_compiler.compile(file_path)
 
         # submit job
-        run_id = self._submit_job(file_path, cluster_spec)
+        run_id = self.api_client.job_runs.submit(
+            job_config.run_name, job_config.job_spec, **job_config.additional_job_config
+        )
+        self.tracker.insert_run_id(run_id)
```

Comment:
should it delete the file if submit_job fails?

Reply:
I'm on the fence about this; users use the uploaded notebook as part of debugging in case of failure.  They can find the same information in the dbt.log, but along with a lot of other information that may be distracting.